cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb

```diff
@@ -0,0 +1,1143 @@
+        "# You may obtain a copy of the License at\n",
+        "#\n",
+        "#     https://www.apache.org/licenses/LICENSE-2.0\n",
+        "#\n",
+        "# Unless required by applicable law or agreed to in writing, software\n",
+        "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+        "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+        "# See the License for the specific language governing permissions and\n",
+        "# limitations under the License."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "C5SwRvvKJvcz"
+      },
+      "source": [
+        "# **Building AI-powered data-driven applications using pgvector, LangChain and LLMs**\n",
+        "\n",
+        "---"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "27wCSEVJhmjp"
+      },
+      "source": [
+        "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/GoogleCloudPlatform/python-docs-samples/blob/main/cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_GhFgSOldCvV"
+      },
+      "source": [
+        "This hands-on tutorial will show you how you can add Generative AI features to your own applications with just a few lines of code using pgvector, LangChain and LLMs on Google Cloud.\n",
+        "\n",
+        "We will build together a sample Python application that will be able to understand and respond to human language queries about the relational data stored in your PostgreSQL database. In fact, we will further push the creative limits of the application by teaching it to generate new content based on our existing dataset.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "93QUUCrvKm03"
+      },
+      "source": [
+        "## Objective\n",
+        "\n",
+        "After completing the steps in this notebook:\n",
+        "- You will have a good understanding of how to use the [pgvector extension](https://github.com/pgvector/pgvector) to store and search vector embeddings in PostgreSQL. Learn more about [vector embeddings](https://cloud.google.com/blog/topics/developers-practitioners/meet-ais-multitool-vector-embeddings).\n",
+        "- You will get a hands-on experience with using the open-source [LangChain framework](https://python.langchain.com/en/latest/index.html) to develop applications powered by large language models. LangChain makes it easier to develop and deploy applications against any LLM model in a vendor-agnostic manner.\n",
+        "- You will learn about the powerful features in [Google PaLM models made available through Vertex AI](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/**overview**).\n"
```

Comment:
https://cloud.google.com/vertex-ai/docs/generative-ai/learn/**overview** gives 404

Reply:
@saketj looks like url works without the `**`s in it: https://cloud.google.com/vertex-ai/docs/generative-ai/learn/overview

corresponding line: https://github.com/saketj/python-docs-samples/blob/main/cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb?short_path=92f0432#L88

Reply:
Thanks for catching this. The ** were left overs from previous edit. Fixed the URL now.